modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
+                        .build(), true);
 
-    public CouchbaseContainer withIndex(boolean index) {
-        this.index = index;
-        return self();
+                    checkSuccessfulResponse(queryResponse, "Could not create primary index for bucket " + bucket.getName());
+                } else {
+                    logger().info("Primary index creation for bucket " + bucket.getName() + " ignored, since QUERY service is not present.");
+                }
+            }
+        }
     }
 
-    public CouchbaseContainer withPrimaryIndex(boolean primaryIndex) {
-        this.primaryIndex = primaryIndex;
-        return self();
+    /**
+     * Helper method to extract the internal IP address based on the network configuration.
+     */
+    private String getInternalIpAddress() {
+        final Map<String, ContainerNetwork> networks = getContainerInfo().getNetworkSettings().getNetworks();
+        for (ContainerNetwork network : networks.values()) {
+            return network.getIpAddress();
+        }
+        throw new IllegalStateException("No network available to extract the internal IP from!");
     }
 
-    public CouchbaseContainer withFts(boolean fts) {
-        this.fts = fts;
-        return self();
+    /**
+     * Helper method to check if the response is successful and release the body if needed.
+     *
+     * @param response the response to check.
+     * @param message the message that should be part of the exception of not successful.
+     */
+    private void checkSuccessfulResponse(final Response response, final String message) {
+        try {
+            if (!response.isSuccessful()) {
+                throw new IllegalStateException(message + ": " + response.toString());
+            }
+        } finally {
+            if (response.body() != null) {
+                response.body().close();
+            }
+        }
     }
 
-    @Getter
-    @RequiredArgsConstructor
-    protected enum CouchbasePort {
-        REST("rest_port", 8091, true),
-        CAPI("capi_port", 8092, false),
-        QUERY("query_port", 8093, false),
-        FTS("fts_http_port", 8094, false),
-        CBAS("cbas_http_port", 8095, false),
-        EVENTING("eventing_http_port", 8096, false),
-        MEMCACHED_SSL("memcached_ssl_port", 11207, false),
-        MEMCACHED("memcached_port", 11210, false),
-        REST_SSL("ssl_rest_port", 18091, true),
-        CAPI_SSL("ssl_capi_port", 18092, false),
-        QUERY_SSL("ssl_query_port", 18093, false),
-        FTS_SSL("fts_ssl_port", 18094, false),
-        CBAS_SSL("cbas_ssl_port", 18095, false),
-        EVENTING_SSL("eventing_ssl_port", 18096, false);
-
-        final String name;
-
-        final int originalPort;
-
-        final boolean dynamic;
+    /**
+     * Checks if already running and if so raises an exception to prevent too-late setters.
+     */
+    private void checkNotRunning() {
```

Comment:
Nice idea from a user perspective, but this behaves very different (better?) than other TC modules.

Reply:
we also check `isRunning()` from `getMappedPort`, for example

Reply:
@bsideup funny: that's how I ended it up adding it like this, because I accidentally had to get a mapped port before it was running...